python/llm/src/ipex_llm/transformers/npu_models/xlm_mp.py

```diff
@@ -0,0 +1,727 @@
+        return_dict: Optional[bool] = True,
+    ) -> Union[Tuple[torch.Tensor], BaseModelOutputWithPastAndCrossAttentions]:
+
+        all_hidden_states = () if output_hidden_states else None
+        all_self_attentions = () if output_attentions else None
+        all_cross_attentions = () if output_attentions and self.config.add_cross_attention else None
+
+        next_decoder_cache = () if use_cache else None
+        layer_outputs = prefill_runner.forward(hidden_states, attention_mask)
+
+        layer_outputs = layer_outputs.to(torch.float32)
+        hidden_states = layer_outputs
+
+        if use_cache:
+            next_decoder_cache += (layer_outputs[-1],)
+
+        if output_attentions:
+            all_self_attentions = all_self_attentions + (layer_outputs[1],)
+            if self.config.add_cross_attention:
+                all_cross_attentions = all_cross_attentions + (layer_outputs[2],)
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_states,)
+
+        if not return_dict:
+            return tuple(
+                v
+                for v in [
+                    hidden_states,
+                    next_decoder_cache,
+                    all_hidden_states,
+                    all_self_attentions,
+                    all_cross_attentions,
+                ]
+                if v is not None
+            )
+        return BaseModelOutputWithPastAndCrossAttentions(
+            last_hidden_state=hidden_states,
+            past_key_values=next_decoder_cache,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attentions,
+            cross_attentions=all_cross_attentions,
+        )
+
+    return xlm_fused_encoder_forward
+
+
+class XLMPoolLinear(NNFactory):
+    def __init__(
+        self,
+        input_shape,
+        output_channels,
+        input_channels,
+        device: str = "NPU",
+    ):
+        super().__init__(False, device)
+
+        # define input
+        input_node = self.parameter(input_shape, dtype=np.float16)
+        res = self.linear(input_node, output_channels, input_channels, bias=True)
+
+        # define outputs
+        res = self.convert_to_fp16(res)
+
+        print("start compiling")
+        self.compile()
+
+
+class XLMPoolLayer(torch.nn.Module):
+    def __init__(
+        self,
+        weight,
+        bias,
+        output_channel,
+        input_channel,
+    ):
+        super().__init__()
+
+        self.op_id = str(uuid.uuid4())
+        self.parameters = [weight, bias]
+        self.backend_cls_pooler = partial(
+            XLMPoolLinear,
+            output_channels=output_channel,
+            input_channels=input_channel,
+        )
+
+    def forward(self, hidden_states):
+        backend_cls = self.backend_cls_pooler
+        hidden_states = hidden_states.to(torch.float16)
+        return run_model(hidden_states, self.parameters, backend_cls, self.op_id)
+
+
+class LayerNorm(NNFactory):
```

Comment:
This is also used in minicpm-v, maybe we could move it to `common.py` as a common implementatin of `torch.nn.LayerNorm` later.

Reply:
Good idea, will do this in next pr.